File: dataset_reduzido/avatar/Python/Code/codeforces_287_A.py
t = [ ]
for i in range ( 4 ) :
    l = list ( map ( str , input ( ) ) )
    t.append ( l )
black = 0
white = 0
correct = 0
for i in range ( 3 ) :
    for j in range ( 3 ) :
        if t [ i ] [ j ] == '#' :
            black += 1
        else :
            white += 1
        if t [ i ] [ j + 1 ] == '#' :
            black += 1
        else :
            white += 1
        if t [ i + 1 ] [ j ] == '#' :
            black += 1
        else :
            white += 1
        if t [ i + 1 ] [ j + 1 ] == '#' :
            black += 1
        else :
            white += 1
        if ( black == 0 or black == 3 or white == 0 or white == 3 or white == 4 or black == 4 ) :
            correct = 1
            break
        black = 0
        white = 0
if correct == 1 :
    print ( 'YES' )
else :
    print ( 'NO' )

File: output/Llama-2-100/avatar/Python/Java/codeforces_287_A.java
import java.util.ArrayList;
import java.util.List;

public class codeforces_287_A {
    public static void main(String[] args) {
        ArrayList<List<String>> t = new ArrayList<>();
        for (int i = 0; i < 4; i++) {
            List<String> l = new ArrayList<>();
            for (int j = 0; j < 3; j++) {
                String input = JOptionPane.showInputDialog(null, "Enter a character: ");
                l.add(input);
            }
            t.add(l);
        }
        int black = 0;
        int white = 0;
        int correct = 0;
        for (int i = 0; i < 3; i++) {
            for (int j = 0; j < 3; j++) {
                if (t.get(i).get(j).equals("#")) {
                    black++;
                } else {
                    white++;
                }
                if (t.get(i).get(j + 1).equals("#")) {
                    black++;
                } else {
                    white++;
                }
                if (t.get(i + 1).get(j).equals("#")) {
                    black++;
                } else {
                    white++;
                }
                if (t.get(i + 1).get(j + 1).equals("#")) {
                    black++;
                } else {
                    white++;
                }
                if (black == 0 || black == 3 || white == 0 || white == 3 || white == 4 || black == 4) {
                    correct++;
                    break;
                }
                black = 0;
                white = 0;
            }
        }
        if (correct == 1) {
            JOptionPane.showMessageDialog(null, "YES");
        } else {
            JOptionPane.showMessageDialog(null, "NO");
        }
    }
}


